SeaPublicWebsite/Services/RecommendationService.cs

```diff
@@ -1,130 +1,433 @@
-            if (userData.FlatType != FlatType.GroundFloor && userData.FlatType != FlatType.MiddleFloor
-                && userData.RoofConstruction != RoofConstruction.Flat
-                && userData.AccessibleLoftSpace == AccessibleLoftSpace.Yes
-                && userData.RoofInsulated != RoofInsulated.Yes
-                && !UserDataHelper.HasRoofInsulation(userData))
+        private static int GetConvertedWallType(WallConstruction? wallConstruction,
+            SolidWallsInsulated? solidWallsInsulated,
+            CavityWallsInsulated? cavityWallsInsulated)
+        {
+            return wallConstruction switch
             {
-                userRecommendationKeys.Add(RecommendationKey.AddLoftInsulation);
-            }
+                WallConstruction.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                WallConstruction.Solid => solidWallsInsulated switch
+                {
+                    SolidWallsInsulated.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                    SolidWallsInsulated.No => (int) WallTypeEnum.SolidWallsWithoutInsulation,
+                    SolidWallsInsulated.Some => (int) WallTypeEnum.SolidWallsWithoutInsulation,
+                    SolidWallsInsulated.All => (int) WallTypeEnum.SolidWallsWithInsulation,
+                    _ => throw new ArgumentOutOfRangeException()
+                },
+                WallConstruction.Cavity => cavityWallsInsulated switch
+                {
+                    CavityWallsInsulated.DoNotKnow => (int) WallTypeEnum.DontKnow,
+                    CavityWallsInsulated.No => (int) WallTypeEnum.CavityWallsWithoutInsulation,
+                    CavityWallsInsulated.Some => (int) WallTypeEnum.CavityWallsWithoutInsulation,
+                    CavityWallsInsulated.All => (int) WallTypeEnum.CavityWallsWithInsulation,
+                    _ => throw new ArgumentOutOfRangeException()
+                },
+                WallConstruction.Mixed =>
+                    //assumption:
+                    (int) WallTypeEnum.DontKnow,
+                WallConstruction.Other => (int) WallTypeEnum.DontKnow,
+                _ => throw new ArgumentOutOfRangeException()
+            };
+        }
 
-            // Not for mid or top floor flat, not for solid floors or properties with insulation OR don't know and > 1996
-            if (userData.FlatType != FlatType.TopFloor && userData.FlatType != FlatType.MiddleFloor
-                && UserDataHelper.HasSuspendedFloor(userData)
-                && !UserDataHelper.HasInsulatedFloor(userData))
+        private static int GetConvertedRoofType(RoofConstruction? roofConstruction, RoofInsulated? roofInsulated)
+        {
+            return roofConstruction switch
             {
-                userRecommendationKeys.Add(RecommendationKey.GroundFloorInsulation);
-            }
+                RoofConstruction.Flat =>
+                    //assumption:
+                    (int) RoofTypeEnum.DontKnow,
```

Comment:
Not your fault, but this seems really weird. Why are we asking about flat roofs if we can't do anything with the information?

Reply:
Could you raise this with Dharmesh please, just so it doesn't get forgotten about.